// [[Reason: Invalid computed property can only be detected by TSC]] ////

//// [tests/cases/compiler/decoratorsOnComputedProperties.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -15,8 +15,11 @@
     ["property3"]: any;
     [Symbol.isConcatSpreadable]: any;
     ["property4"]: any;
     [Symbol.match]: any;
+    [fieldNameA]: any;
+    [fieldNameB]: any;
+    [fieldNameC]: any;
 }
 declare class C {
     ["property"]: any;
     [Symbol.toStringTag]: any;
@@ -25,8 +28,11 @@
     ["property3"]: any;
     [Symbol.isConcatSpreadable]: any;
     ["property4"]: any;
     [Symbol.match]: any;
+    [fieldNameA]: any;
+    [fieldNameB]: any;
+    [fieldNameC]: any;
 }
 declare class E {
     ["property"]: any;
     [Symbol.toStringTag]: any;
@@ -35,8 +41,11 @@
     ["property3"]: any;
     [Symbol.isConcatSpreadable]: any;
     ["property4"]: any;
     [Symbol.match]: any;
+    [fieldNameA]: any;
+    [fieldNameB]: any;
+    [fieldNameC]: any;
 }
 declare class G {
     ["property"]: any;
     [Symbol.toStringTag]: any;
@@ -45,8 +54,11 @@
     ["property3"]: any;
     [Symbol.isConcatSpreadable]: any;
     ["property4"]: any;
     [Symbol.match]: any;
+    [fieldNameA]: any;
+    [fieldNameB]: any;
+    [fieldNameC]: any;
 }
 declare class I {
     ["property"]: any;
     [Symbol.toStringTag]: any;
@@ -55,8 +67,11 @@
     ["property3"]: any;
     [Symbol.isConcatSpreadable]: any;
     ["property4"]: any;
     [Symbol.match]: any;
+    [fieldNameA]: any;
+    [fieldNameB]: any;
+    [fieldNameC]: any;
 }
 
 /// [Errors] ////
 
@@ -64,13 +79,10 @@
 decoratorsOnComputedProperties.ts(18,5): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(19,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(20,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(21,5): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(21,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(22,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(22,8): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(23,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(23,8): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(27,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(28,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(29,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(30,5): error TS1206: Decorators are not valid here.
@@ -83,13 +95,10 @@
 decoratorsOnComputedProperties.ts(52,5): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(53,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(54,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(55,5): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(55,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(56,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(56,8): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(57,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(57,8): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(62,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(63,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(64,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(65,5): error TS1206: Decorators are not valid here.
@@ -102,13 +111,10 @@
 decoratorsOnComputedProperties.ts(88,5): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(89,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(90,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(92,5): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(92,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(93,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(93,8): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(94,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(94,8): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(98,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(99,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(100,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(101,5): error TS1206: Decorators are not valid here.
@@ -121,13 +127,10 @@
 decoratorsOnComputedProperties.ts(124,5): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(125,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(126,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(128,5): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(128,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(129,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(129,8): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(131,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(131,8): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(135,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(136,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(137,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(138,5): error TS1206: Decorators are not valid here.
@@ -140,13 +143,10 @@
 decoratorsOnComputedProperties.ts(162,5): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(163,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(164,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
 decoratorsOnComputedProperties.ts(166,5): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(166,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(167,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(167,8): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(169,8): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-decoratorsOnComputedProperties.ts(169,8): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 decoratorsOnComputedProperties.ts(173,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(174,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(175,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(176,5): error TS1206: Decorators are not valid here.
@@ -158,9 +158,9 @@
 decoratorsOnComputedProperties.ts(186,5): error TS1206: Decorators are not valid here.
 decoratorsOnComputedProperties.ts(188,5): error TS1206: Decorators are not valid here.
 
 
-==== decoratorsOnComputedProperties.ts (97 errors) ====
+==== decoratorsOnComputedProperties.ts (82 errors) ====
     function x(o: object, k: PropertyKey) { }
              ~
 !!! error TS9007: Function must have an explicit return type annotation with --isolatedDeclarations.
 !!! related TS9031 decoratorsOnComputedProperties.ts:1:10: Add a return type to the function declaration.
@@ -191,20 +191,14 @@
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
         [fieldNameA]: any;
         ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-        ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         @x [fieldNameB]: any;
            ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-           ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         @x [fieldNameC]: any = null;
            ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-           ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
     }
     
     void class B {
         @x ["property"]: any;
@@ -263,20 +257,14 @@
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
         [fieldNameA]: any;
         ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-        ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         @x [fieldNameB]: any;
            ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-           ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         @x [fieldNameC]: any = null;
            ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-           ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         ["some" + "method"]() {}
     }
     
     void class D {
@@ -338,20 +326,14 @@
         ["some" + "method"]() {}
         [fieldNameA]: any;
         ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-        ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         @x [fieldNameB]: any;
            ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-           ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         @x [fieldNameC]: any = null;
            ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-           ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
     }
     
     void class F {
         @x ["property"]: any;
@@ -412,21 +394,15 @@
         ["some" + "method"]() {}
         [fieldNameA]: any;
         ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-        ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         @x [fieldNameB]: any;
            ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-           ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         ["some" + "method2"]() {}
         @x [fieldNameC]: any = null;
            ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-           ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
     }
     
     void class H {
         @x ["property"]: any;
@@ -488,21 +464,15 @@
         @x ["some" + "method"]() {}
         [fieldNameA]: any;
         ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-        ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         @x [fieldNameB]: any;
            ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-           ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         ["some" + "method2"]() {}
         @x [fieldNameC]: any = null;
            ~~~~~~~~~~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-           ~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
     }
     
     void class J {
         @x ["property"]: any;
